R/stat-smooth.r

```diff
@@ -86,21 +88,25 @@ StatSmooth <- ggproto("StatSmooth", Stat,
 
       if (max_group < 1000) {
         params$method <- "loess"
+        params$formula <- params$formula %||% (y ~ x)
```

Comment:
You're now setting the formula in two places — I don't think you need to set when figuring out which `method` to use, you can just set it once the method is known.

Reply:
If the formula isn't set before the [message()](https://github.com/tidyverse/ggplot2/blob/master/R/stat-smooth.r#L93) call just below, then the message could include a `NULL` instead of the relevant formula, breaking [this test](https://github.com/tidyverse/ggplot2/blob/master/tests/testthat/test-geom-smooth.R#L27) for example.  Maybe a solution could be to have a logical keep track of whether or not `params$method` is originally `"auto"` and then move the `message()` call down until after the formula has been set once and wrap it in `if (using_auto_method)`, although that doesn't seem great to me.

Reply:
Oh, I see the problem — now that `method` and `formula` are being set independently, you need to break that message up into two pieces. You should only ever inform the user when the parameter was chosen automatically, not when they supplied it.